src/context/AuthContext.jsx

```diff
@@ -0,0 +1,148 @@
+import axios from "axios";
+import React, { useEffect, useState, createContext, useContext } from "react";
+import { useNavigate } from "react-router-dom";
+import { useToast } from "./ToastContext";
+
+const AuthContext = createContext();
+const initialState = {
+  loading: true,
+  data: [],
+  error: "",
+};
+const AuthProvider = ({ children }) => {
+  const { setToast } = useToast();
+  const navigate = useNavigate();
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
```

Comment:
suggestion: you could handel logged in user with just user, instead of using a different state